zabbix_auto_config/processing.py

```diff
@@ -666,11 +677,27 @@ def do_update(self):
             # if api ver >= 4.4
             if "inventory_mode" in zabbix_host.keys():
                 if zabbix_host["inventory_mode"] != "1":
-                    self.set_inventory(zabbix_host, 1)
+                    self.set_inventory_mode(zabbix_host, 1)
             # else api ver. < 4.4
             else:
                 if zabbix_host["inventory"]["inventory_mode"] != "1":
-                    self.set_inventory(zabbix_host, 1)
+                    self.set_inventory_mode(zabbix_host, 1)
+
+            if db_host["inventory"]:
+                if zabbix_host["inventory"]:
+                    inventories_to_add = dict(set(db_host["inventory"].items()).difference(set(zabbix_host["inventory"].items())))
+                else:
+                    inventories_to_add = db_host["inventory"]
```

Comment:
I think I'd rather see if values from db_host actually changes the zabbix_host. E.g don't do the set conversion and back to dict. I think that would be easier to read as well.

But this works as well as far as I can tell.

Reply:
To clarify with an example:

Something like this could be done:
```python
changed_inventory = {k: v for k, v in db_host["inventory"].items() if db_host["inventory"][k] != zabbix_host["inventory"].get(k, None)}
```

I believe this is a bit easier to read/more intuitive for the reader. It removes the conversions.